src/main/java/com/softserve/academy/spaced/repetition/config/PointsBalanceEvent.java

```diff
@@ -0,0 +1,22 @@
+package com.softserve.academy.spaced.repetition.config;
+
+import com.softserve.academy.spaced.repetition.domain.User;
+import org.springframework.context.ApplicationEvent;
+
+public class PointsBalanceEvent extends ApplicationEvent {
+
+    private User user;
+
+    public PointsBalanceEvent(Object source, User user) {
+        super(source);
+        this.user = user;
+    }
+
+    public User getUser() {
+        return user;
+    }
+
+    public void setUser(User user) {
```

Comment:
Please remove the setter to make User required.

Reply:
Thanks for comments. You are right, there was an error in test. I fixed it and removed setter from event.